```
golangci-lint run --allow-parallel-runners
traces_exporter.go:108:3: (*traceExporter).pushTraceData - result 0 (error) is always nil (unparam)
) error {
  ^
make[2]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/exporter/datadogexporter'
make[2]: *** [../../Makefile.Common:95: lint] Error 1
make[1]: *** [Makefile:148: for-all-target-./exporter/datadogexporter] Error 2
```

--- a/exporter/datadogexporter/internal/scrub/scrub.go
+++ b/exporter/datadogexporter/internal/scrub/scrub.go
@@ -16,7 +16,6 @@ package scrub
 
 import (
 	"regexp"
-	"strings"
 )
 
 // ErrorScrubber scrubs error from sensitive details
@@ -28,7 +27,6 @@ type Scrubber interface {
 // replacer structure to store regex matching and replacement functions
 type replacer struct {
 	Regex *regexp.Regexp
-	Hints []string // If any of these hints do not exist in the line, then we know the regex wont match either
 	Repl  string
 }
 
@@ -61,13 +59,11 @@ func NewScrubber() Scrubber {
 			{
 				// If hinted, mask the value regardless if it doesn't match 32-char hexadecimal string
 				Regex: regexp.MustCompile(`(api_?key=)\b[a-zA-Z0-9]+([a-zA-Z0-9]{5})\b`),
-				Hints: []string{"api_key", "apikey"},
 				Repl:  `$1***************************$2`,
 			},
 			{
 				// If hinted, mask the value regardless if it doesn't match 40-char hexadecimal string
 				Regex: regexp.MustCompile(`(ap(?:p|plication)_?key=)\b[a-zA-Z0-9]+([a-zA-Z0-9]{5})\b`),
-				Hints: []string{"app_key", "appkey", "application_key"},
 				Repl:  `$1***********************************$2`,
 			},
 			{
@@ -92,16 +88,7 @@ func (s *scrubber) Scrub(err error) error {
 // Scrub sensitive details from a string.
 func (s *scrubber) scrubStr(data string) string {
 	for _, repl := range s.replacers {
-		containsHint := false
-		for _, hint := range repl.Hints {
-			if strings.Contains(data, hint) {
-				containsHint = true
-				break
-			}
-		}
-		if len(repl.Hints) == 0 || containsHint {
-			data = repl.Regex.ReplaceAllString(data, repl.Repl)
-		}
+		data = repl.Regex.ReplaceAllString(data, repl.Repl)
 	}
 	return data
 }

--- a/exporter/datadogexporter/internal/scrub/scrub_test.go
+++ b/exporter/datadogexporter/internal/scrub/scrub_test.go
@@ -63,14 +63,15 @@ func TestScrubber(t *testing.T) {
 	scrubber := NewScrubber()
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			assert.Equal(t, test.expected, scrubber.Scrub(errors.New(test.original)).Error())
+			assert.EqualError(t, scrubber.Scrub(errors.New(test.original)), test.expected)
 		})
 	}
 }
 
 func TestPermanentErrorScrub(t *testing.T) {
-	err := consumererror.NewPermanent(errors.New("this is an error"))
+	err := consumererror.NewPermanent(errors.New("this is an error with an app key AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABBBB"))
 	scrubber := NewScrubber()
 	err = scrubber.Scrub(err)
 	assert.True(t, consumererror.IsPermanent(err))
+	assert.EqualError(t, err, "Permanent error: this is an error with an app key ***********************************ABBBB")
 }

--- a/exporter/datadogexporter/traces_exporter.go
+++ b/exporter/datadogexporter/traces_exporter.go
@@ -22,6 +22,7 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/trace/exportable/obfuscate"
 	"github.com/DataDog/datadog-agent/pkg/trace/exportable/pb"
 	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumerhelper"
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
 	"gopkg.in/zorkian/go-datadog-api.v2"
@@ -101,6 +102,9 @@ func newTracesExporter(ctx context.Context, params component.ExporterCreateSetti
 func (exp *traceExporter) pushTraceDataScrubbed(ctx context.Context, td pdata.Traces) error {
 	return exp.scrubber.Scrub(exp.pushTraceData(ctx, td))
 }
+
+// force pushTraceData to be a ConsumeTracesFunc, even if no error is returned.
+var _ consumerhelper.ConsumeTracesFunc = (*traceExporter)(nil).pushTraceData
 
 func (exp *traceExporter) pushTraceData(
 	ctx context.Context,

